packages/@lwc/ssr-compiler/src/compile-template/transformers/component/slotted-content.ts

```diff
@@ -246,24 +256,65 @@ export function getSlottedContent(
             bAddSlottedContent(
                 slotName,
                 boundVariable,
-                irChildrenToEs(child.children, cxt),
+                optimizeAdjacentYieldStmts(irChildrenToEs(child.children, cxt)),
                 b.identifier('scopedSlottedContentMap')
             )
         );
         cxt.popLocalVars();
         return addLightContentExpr;
     });
 
-    const hasShadowSlottedContent = b.literal(shadowSlotContent.length > 0);
-    const hasLightSlottedContent = b.literal(lightSlotContent.length > 0);
-    const hasScopedSlottedContent = b.literal(scopedSlotContent.length > 0);
+    const hasShadowSlottedContent = shadowSlotContent.length > 0;
+    const hasLightSlottedContent = lightSlotContent.length > 0;
+    const hasScopedSlottedContent = scopedSlotContent.length > 0;
     cxt.isSlotted = isSlotted;
 
+    if (hasShadowSlottedContent || hasLightSlottedContent || hasScopedSlottedContent) {
+        cxt.import('addSlottedContent');
+    }
+
+    // Elsewhere, nodes and their subtrees are cloned. This design decision means that
+    // the node objects themselves cannot be used as unique identifiers (e.g. as keys
+    // in a map). However, for a given template, a node's location information does
+    // uniquely identify that node.
+    const uniqueNodeId = `${node.name}:${node.location.start}:${node.location.end}`;
+
+    const localVars = cxt.getLocalVars();
+    const localVarIds = localVars.map(b.identifier);
+
+    if (hasShadowSlottedContent && !cxt.slots.shadow.isDuplicate(uniqueNodeId)) {
+        // Colon characters in <lwc:component> element name will result in an invalid
+        // JavaScript identifier if not otherwise accounted for.
+        const kebabCmpName = kebabCaseToCamelCase(node.name).replace(':', '_');
```

Comment:
```suggestion
        // Colon characters in <lwc:component> element name will result in an invalid
        // JavaScript identifier if not otherwise accounted for.
        const kebabCmpName = kebabCaseToCamelCase(node.name).replace(/\W/g, '_');
```
Do we need to generalize this?

Reply:
I considered it. So far this is the only place we need to do something like this. I don't think it would be useful outside of the SSR template compiler at all, so doesn't belong in `@lwc/shared` (which was my first instinct). The only place I thought it could go is `packages/@lwc/ssr-compiler/src/compile-template/shared.ts`. I'll just do that.